main.tf

```diff
@@ -1,4 +1,53 @@
 
+# ECS Service
+resource "aws_ecs_service" "service" {
+  count                              = var.deploy_service ? 1 : 0
+  name                               = "${var.name}_service"
+  cluster                            = var.cluster
+  task_definition                    = aws_ecs_task_definition.this[0].id
+  desired_count                      = var.desired_count
+  deployment_minimum_healthy_percent = var.tasks_minimum_healthy_percent
+  deployment_maximum_percent         = var.tasks_maximum_percent
+  launch_type                        = var.launch_type
+
+  dynamic "network_configuration" {
+    for_each = var.network_mode == "awsvpc" ? [var.network_configuration] : []
+    content {
+      subnets          = lookup(network_configuration.value, "subnets", null)
+      assign_public_ip = lookup(network_configuration.value, "assign_public_ip", null)
+      security_groups  = lookup(network_configuration.value, "security_groups", [join("", aws_security_group.service.*.id)])
+    }
+  }
+
+  dynamic "load_balancer" {
+    for_each = aws_lb.main
+    content {
+      container_name   = var.container_name
+      container_port   = var.container_port
+      target_group_arn = aws_lb_target_group.main_tg[0].arn
+    }
+  }
+  depends_on = [
+    aws_security_group.service
+  ]
+}
+
```

Comment:
Comparing the example used and the root module, this may need some restructuring

Reply:
> Comparing the example used and the root module, this may need some restructuring
Sorry, not restructuring the module per say, I think only the load balancers block, with option for picking the lb created by module or an external one